cockatrice/src/client/ui/phases_toolbar.cpp

```diff
@@ -74,13 +74,26 @@ void PhaseButton::updateAnimation()
     if (!highlightable)
         return;
 
-    if (active) {
-        if (++activeAnimationCounter >= 10)
-            activeAnimationTimer->stop();
+    // the counter ticks up to 10 when active and down to 0 when inactive
+    if (active && activeAnimationCounter < 10) {
+        ++activeAnimationCounter;
+    } else if (!active && activeAnimationCounter > 0) {
+        --activeAnimationCounter;
     } else {
-        if (--activeAnimationCounter <= 0)
-            activeAnimationTimer->stop();
+        activeAnimationTimer->stop();
     }
+
+    update();
+}
+
+/**
+ * @brief Immediately resets the button to the inactive state, without going through the animation.
+ */
+void PhaseButton::reset()
+{
+    activeAnimationTimer->stop();
+    active = false;
+    activeAnimationCounter = highlightable ? 0 : 9;
```

Comment:
I think this'd read better as a simple if statement
```suggestion
    if (highlightable) {
        activeAnimationCounter = 0;
    }
```